docs/developer/architecture/alerting.rst

```diff
@@ -43,3 +43,295 @@ maintenance operation.
+
+.. _PromQL: https://prometheus.io/docs/prometheus/latest/querying/basics/
+
+Edit Existing Prometheus Alert Rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To edit existing :term:`Prometheus` rules, we can't only define new
+``PrometheusRules`` resources since :term:`Prometheus` Operator will not
+overwrite those already existing, but will rather append them to the list of
+rules, ending up with 2 rules with the same name but different parameters.
+
+We also can't edit the ``PrometheusRules`` deployed by MetalK8s, otherwise we
+would lose these changes in case of cluster restoration, upgrade or downgrade.
+
+So, in order to allow the user to customize the alert rules, we will pick up
+some of them (the most relevant ones) and expose only few parts of their
+configurations (e.g. threshold) to be customized.
+
+It also makes the customization of these alert rules easier for the user as,
+for example, he will not need to understand PromQL_ to adapt the threshold of
+an alert rule.
+
+Since in :term:`Prometheus` rules, there are duplicated group name + alert rule
+name, we also need to take the severity into account to understand which
+specific alert we're editing.
+
+These customization will be stored in the ``metalk8s-prometheus-config``
+:term:`ConfigMap` with something like::
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: metalk8s-prometheus-config
+      namespace: metalk8s-monitoring
+    data:
+      config.yaml: |-
+        apiVersion: addons.metalk8s.scality.com
+        kind: PrometheusConfig
+        spec:
+          deployment:
+            replicas: 1
+          rules:
+            <alertGroupName>:
+              <alertName>:
+                warning:
+                  threshold: 30
+                critical:
+                  threshold: 10
+            <anotherAlertGroupName>:
+              <anotherAlertName>:
+                critical:
+                  threshold: 20
+                  anotherThreshold: 10
+
+The ``PrometheusRules`` object manifests
+``salt/metalk8s/addons/prometheus-operator/deployed/chart.sls`` need
+to be templatized to consume these customizations through Jinja.
+
+Default values for customizable alert rules to fallback on, if not defined
+in the :term:`ConfigMap`, will be set in ``salt/metalk8s/defaults.yaml``.
+
+Custom Alertmanager Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We will use the already existing ``metalk8s-alertmanager-config``
+:term:`ConfigMap` to store the term:`Alertmanager` configuration
+customizations.
+
+A Salt module will be developed to manipulate this object, so the logic
+can be kept in only one place.
+
+This module must provide necessary methods to show or edit the configuration
+in 2 different ways:
+
+   * simple
+   * advanced
+
+The ``simple`` mode will only display and allow to change some specific
+configuration, such as the receivers or the inhibit rules, and in an as
```

Comment:
What about mute and unmute an alert rule since this should be part of the simple mode??

Reply:
Mute and unmute alerts are not part of the Alertmanager configuration, see `Silence an Alert` part of this document.